rslib/src/search/browser.rs

```diff
@@ -0,0 +1,335 @@
+                "deck".to_string(),
+            ]
+        });
+        let mut context = row_context_from_cid(self, id, note_fields_required(&columns))?;
+        Ok(Row {
+            cells: columns
+                .iter()
+                .map(|column| get_cell(column, &mut context))
+                .collect::<Result<_>>()?,
+            color: get_row_color(&context),
+            font: get_row_font(&context)?,
+        })
+    }
+}
+
+fn get_cell(column: &str, context: &mut RowContext) -> Result<Cell> {
+    Ok(Cell {
+        text: get_cell_text(column, context)?,
+        is_rtl: get_is_rtl(column, context),
+    })
+}
+
+fn get_cell_text(column: &str, context: &mut RowContext) -> Result<String> {
+    Ok(match column {
+        "answer" => answer_str(context)?,
+        "cardDue" => card_due_str(context)?,
+        "cardEase" => card_ease_str(context),
+        "cardIvl" => card_interval_str(context),
+        "cardLapses" => context.card.lapses.to_string(),
+        "cardMod" => context.card.mtime.date_string(context.offset),
+        "cardReps" => context.card.reps.to_string(),
+        "deck" => deck_str(context)?,
+        "note" => context.notetype.name.to_owned(),
+        "noteCrt" => note_creation_str(context),
+        "noteFld" => note_field_str(context),
+        "noteMod" => context.note.mtime.date_string(context.offset),
+        "noteTags" => context.note.tags.join(" "),
+        "question" => question_str(context)?,
+        "template" => template_str(context)?,
+        _ => "".to_string(),
+    })
+}
+
+fn answer_str(context: &RowContext) -> Result<String> {
+    let text = context
+        .answer_nodes
+        .as_ref()
+        .unwrap()
+        .iter()
+        .map(|node| match node {
+            RenderedNode::Text { text } => text,
+            RenderedNode::Replacement {
+                field_name: _,
+                current_text,
+                filters: _,
+            } => current_text,
+        })
+        .join("");
+    Ok(html_to_text_line(&extract_av_tags(&text, false).0).to_string())
+}
+
+fn card_due_str(context: &mut RowContext) -> Result<String> {
+    Ok(if context.original_deck()?.is_some() {
+        context.i18n.tr(TR::BrowsingFiltered).into()
+    } else if context.card.queue == CardQueue::New || context.card.ctype == CardType::New {
+        context.i18n.trn(
+            TR::StatisticsDueForNewCard,
+            tr_args!["number"=>context.card.due],
+        )
+    } else {
+        let date = match context.card.queue {
+            CardQueue::Learn => TimestampSecs(context.card.due as i64),
+            CardQueue::DayLearn | CardQueue::Review => TimestampSecs::now().adding_secs(
+                ((context.card.due - context.timing.days_elapsed as i32) * 86400) as i64,
+            ),
+            _ => return Ok("".into()),
+        };
+        date.date_string(context.offset)
+    })
+}
+
+fn card_ease_str(context: &RowContext) -> String {
+    match context.card.ctype {
```

Comment:
All of these routines have a RowContext in common. Rather than having them as free-standing functions that take that argument, it might be better to just wrap them all in an impl RowContext and have them take self instead.